pkg/daemon/iptables/iptables.go

```diff
@@ -428,16 +441,33 @@ func generateVxlanFilterRuleSpec(vxlanIf, allIPSet string, protocol Protocol) []
 		"-j", "REJECT", "--reject-with", rejectWithOption(protocol)}
 }
 
+// pod -> node traffic cannot be made to overlay by iptables (because of the DNAT operation on service datapath)
+// this rule is used to make sure the connection from node to pod is ok.
 func generateVxlanPodToNodeReplyMarkRuleSpec(overlayNetSet, nodeIPSet string) []string {
 	return []string{"-A", ChainHybridnetPreRouting, "-m", "comment", "--comment", `"mark overlay pod -> node back traffic"`,
 		"-m", "addrtype", "!", "--dst-type", "LOCAL",
 		"-m", "set", "--match-set", overlayNetSet, "src",
 		"-m", "set", "--match-set", nodeIPSet, "dst",
 		"-m", "conntrack", "!", "--ctstate", "NEW,INVALID,DNAT,SNAT",
+		"-j", ChainHybridnetPodToNodeTrafficMark,
+	}
+}
+
+func generatePodToNodeMarkRuleSpec() []string {
+	return []string{"-A", ChainHybridnetPodToNodeTrafficMark, "-m", "comment", "--comment", `"do pod -> node traffic mark"`,
 		"-j", "MARK", "--set-xmark", fmt.Sprintf("%s/%s", PodToNodeBackTrafficMarkString, PodToNodeBackTrafficMarkString),
 	}
 }
 
+// if pod -> node traffic is for svc with "externalTrafficPolicy: Local" do not mark it as overlay
+// TODO: this need to be retained if we use node ip routes to build the pod -> node traffic
+func generateLocalDNATedSkipRuleSpec(localNodeIP net.IP) []string {
```

Comment:
why not use ipset for local node IPs? maybe we can squash several rules into one

Reply:
Because ipset can only be used to match destination or source for a packet, it cannot be used together with `--ctorigdst` function.